app.js

```diff
@@ -1,18 +1,103 @@
-require('dotenv').config();
-
-const express = require('express');
-const hbs = require('hbs');
-
-// require spotify-web-api-node package here:
+require("dotenv").config();
 
+const express = require("express");
 const app = express();
+const hbs = require("hbs");
+// require spotify-web-api-node package here:
+const SpotifyWebApi = require("spotify-web-api-node");
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
+
+//
 
 // Our routes go here:
+app.get("/", (req, res) => {
+  res.render("index");
+});
+
+app.get("/artist-search", (req, res) => {
+ 
+  spotifyApi
+    .searchArtists(req.query.title)
+    .then((data) => {
+      console.log("The received data from the API: ", data.body.artists.items);
+      res.render("artist-search-results", {artistArr: data.body.artists.items});
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+}); 
+app.get('/albums/:artistId', (req, res, next) => {
+
+  spotifyApi.getArtistAlbums(
+      req.params.artistId, {
+          limit: 10,
+          offset: 20
+      },
+      
+      function (err, data) {
+          
+          if (err) {
+              console.error('Something went wrong!', err);
+          } else {
+              const spotifyAlbumsArray = data.body.items;
+
+              const albumsArray = spotifyAlbumsArray.map(function (album) {
+                  return {
+                      name: album.name,
+                      image: album.images[0],
+                      id: album.id
+                  };
+              });
```

Comment:
I liked this approach, that you are organizing the data before sending it to the view.